981.py

```diff
@@ -0,0 +1,32 @@
+class TimeMap:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.store = collections.defaultdict(list)
+
+
+    def set(self, key: str, value: str, timestamp: int) -> None:
+        self.store[key].append((timestamp, value))
+
+
+    def get(self, key: str, timestamp: int) -> str:
+        values = self.store[key]
+        left, right = 0, len(values)
+
+        while left < right:
```

Comment:
LGTM. If permitted by the interviewer, you could also use the `bisect.bisect_right()` [function](https://docs.python.org/3/library/bisect.html#bisect.bisect_right) to replace your own binary search implementation.